packages/web/src/pages/visualizer/VisualizerProvider.tsx

```diff
@@ -98,7 +98,15 @@ const Visualizer = ({
 
   // Rebind audio
   useEffect(() => {
-    if (audioPlayer.audioCtx && playing) Visualizer1?.bind(audioPlayer)
+    if (playing) {
+      if (audioPlayer.audioCtx) {
+        Visualizer1?.bind(audioPlayer)
+      } else {
+        audioPlayer.audio.addEventListener('canplay', () => {
```

Comment:
So is this for mp3 part of it basically?

Reply:
This makes sure the stuff is all connected before the visualizer tries to bind otherwise it may attach an analyser node to nothing. This is solved by a setTimeout too if we want. LMK if you have thoughts on a nicer way

Reply:
No, I think this makes sense. Was just curious if there could be a race condition since both sides are being done in `'canplay'` listeners.